scripts/node_pair_extraction_helpers.py

```diff
@@ -3,6 +3,62 @@
 from file_helpers import *
 from ortholog_helpers import *
 
+def add_valid_edges(list_of_species_nodes, lines):
+    num_edges = 0
+    edges = set()
+    for i in list_of_species_nodes:
+        if(i[0] in lines and i[1] in lines):
+            num_edges += 1
+            edges.add((i[0], i[1]))
+    return [edges, num_edges]
+
+def bifurcate_mapping_into_dictionaries(preprocessed_nodes, left_node_frequencies, right_node_frequencies):
+    counter = 0
+    for i in preprocessed_nodes[0]:
+        if i != "0" and preprocessed_nodes[1][counter] != "0":
+            left_node_frequencies[i] += 1
+            right_node_frequencies[preprocessed_nodes[1][counter]] += 1
+            counter += 1
+    return [left_node_frequencies, right_node_frequencies]
+
+def remove_duplicate_mappings_from_dictionary(preprocessed_nodes, left_node_frequencies, right_node_frequencies):
+    processed_list_of_nodes = list()
+    counter = 0
+    for i in preprocessed_nodes[0]:
+        if i != "0" and preprocessed_nodes[1][counter] != "0":
+            if not(left_node_frequencies[i] >= 1 or int(right_node_frequencies[preprocessed_nodes[1][counter]]) >= 1):
+                processed_list_of_nodes.append((i, preprocessed_nodes[1][counter]))
+        counter += 1
+    return processed_list_of_nodes
+
+def remove_dupes(list_of_nodes):
```

Comment:
if the above is "remove dupes from dictionary", you should name this "remove dupes from list"